internal/datastore/crdb/crdb_test.go

```diff
@@ -151,27 +151,29 @@ func TestCRDBDatastoreWithFollowerReads(t *testing.T) {
 					DebugAnalyzeBeforeStatistics(),
 					WithAcquireTimeout(5*time.Second),
 				)
-				require.NoError(err)
+				require.NoError(t, err)
 				return ds
 			})
 			t.Cleanup(func() {
-				require.NoError(ds.Close())
+				_ = ds.Close()
 			})
 
-			r, err := ds.ReadyState(ctx)
-			require.NoError(err)
-			require.True(r.IsReady)
+			require.EventuallyWithT(t, func(c *assert.CollectT) {
+				r, err := ds.ReadyState(ctx)
+				require.NoError(c, err)
+				require.True(c, r.IsReady, "datastore not ready: %s", r.Message)
+			}, 3*time.Second, 50*time.Millisecond)
```

Comment:
`EventuallyWithT` is kinda nifty - it lets you write a set of assertions using `require` and keep asserting them until they pass. In this case, we're using it because it takes time for the CRDB connections to become both available and idle - previously this test was passing because there was a single available idle connection and the rest were constructing.

This also provides a bit of anecdotal support for the motivation of this work.